scripts/sample_qc.py

```diff
@@ -0,0 +1,332 @@
+                "PC-relate requires SSDs and doesn't work with preemptible workers!"
+            )
+            relatedness_ht = compute_relatedness(
+                data_type,
+                overwrite=args.overwrite,
+            )
+        else:
+            relatedness_ht = hl.read_table(
+                get_ccdg_results_path(data_type=data_type, result="relatedness")
+            ).checkpoint(
+                "gs://ccdg/tmp/relatedness_ht_checkpoint.ht", overwrite=True
+            )  # Copy HT to temp location to overwrite annotation
+        relatedness_ht = annotate_relatedness(
+            relatedness_ht,
+            first_degree_kin_thresholds=tuple(args.first_degree_kin_thresholds),
+            second_degree_min_kin=args.second_degree_kin_cutoff,
+            ibd0_0_max=args.ibd0_0_max,
+        )
+        relatedness_ht.write(
+            get_ccdg_results_path(data_type=data_type, result="relatedness"),
+            overwrite=args.overwrite,
+        )
+
+    if args.compute_related_samples_to_drop:
+        relatedness_ht = hl.read_table(
+            get_ccdg_results_path(data_type=data_type, result="relatedness")
+        )
+        related_samples_to_remove = hl.maximal_independent_set(
+            relartedness_ht.i, pairs.j, False
+        ).checkpoint(
+            get_ccdg_results_path(data_type=data_type, result="related_samples"),
+            overwrite=args.overwrite,
+        )
+
+    if args.update_variant_filtered_pca_mt:
+        pca_var_ht = hl.read_table(get_pca_variants_path())
+        mt = hl.vds.to_dense_mt(get_qc_vds(data_type, split=True))
+        mt = mt.filter_rows(hl.is_defined(pca_var_ht[mt.row_key])).checkpoint(
+            get_pca_variants_path(ld_pruned=True, data=f"ccdg_{data_type}", mt=True),
+            overwrite=args.overwrite,
+            _read_if_exists=(not args.overwrite),
+        )
+
+    if args.run_pc_project:
+        ## TODO: Rank samples and hard filter samples
+        mt = hl.read_matrix_table(
+            get_pca_variants_path(ld_pruned=True, data=f"ccdg_{data_type}", mt=True)
+        )
+        related_ht = hl.read_table(
+            get_ccdg_results_path(data_type=data_type, result="related_samples")
+        )
+
+        related_mt = mt.filter_cols(hl.is_defined(related_mt[mt.col_key]), keep=True)
+        pca_mt = mt.filter_cols(hl.is_defined(related_mt[mt.col_key]), keep=False)
+        pca_loadings = hl.read_table(
+            "gs://gnomad-public-requester-pays/release/3.1/pca/gnomad.v3.1.pca_loadings.ht"
+        )
+
+        pca_ht = hl.experimental.pc_project(
+            pca_mt, pca_loadings.loadings, pca_loadings.pca_af
+        )
+        pca_mt = pca_mt.annotate_cols(scores=pca_ht[pca_mt.col_key].scores)
+
+        related_ht = hl.experimental.pc_project(
+            related_mt, pca_loadings.loadings, pca_loadings.pca_af
+        )
+        related_mt = related_mt.annotate_cols(
+            scores=related_ht[related_mt.col_key].scores
+        )
```

Comment:
If we're going the projection route you probably don't need to project relateds and unrelateds separately. But fine to keep this so that you can easily bifurcate the process to run PCA natively on this dataset if you want. Also this just sort of ends?